### PR TITLE
594 fix debounce on all input blocks

### LINKED ISSUE #16: [TASK] Fix debounce on all Input Blocks
look through all input blocks and have debounce occur onChange versus on data.value change

### PR DESCRIPTION
**Pull request by phkem**

### CODE REVIEW

**Comment by johbaxter** [comment]
Will review by Friday

**Comment by johbaxter** [comment]
Resolves #16

**Comment by phkem** [comment]
Resolves #16

**Comment by phkem** [comment]
Resolves #16

File: packages/client/src/components/block-defaults/checkbox-block/CheckboxBlock.tsx [hunk 1]
```diff
@@ -52,6 +46,9 @@ export const CheckboxBlock: BlockComponent = observer(({ id }) => {
                     const value = e.target.checked;
                     // update the value
                     setData('value', value);
+                    debounce(() => {
```
Comment (neelneelneel) [inline packages/client/src/components/block-defaults/checkbox-block/CheckboxBlock.tsx]: This violates the rules of Hooks. See https://react.dev/warnings/invalid-hook-call-warning

**Review by neelneelneel — CHANGES_REQUESTED** [review]
Some issues:
- [ ] https://react.dev/warnings/invalid-hook-call-warning

File: packages/client/src/components/block-defaults/upload-block/UploadBlock.tsx [hunk 2]
```diff
@@ -26,6 +28,7 @@ export interface UploadBlockDef extends BlockDef<'upload'> {
 export const UploadBlock: BlockComponent = observer(({ id }) => {
     const { attrs, data, setData, uploadFile, listeners } =
         useBlock<UploadBlockDef>(id);
+    const [changedValue, setChangedValue] = useState(false);
```
Comment (johbaxter) [inline packages/client/src/components/block-defaults/upload-block/UploadBlock.tsx:31]: What is the purpose of this?

**Comment by johbaxter** [comment]
Just took a look again, and noticed the debounce is not triggering the callback within it.  Please test this across all the changed blocks.

**Pull request closed without merging**